crates/matrix-sdk/src/room/mod.rs

```diff
@@ -620,13 +620,26 @@ impl Room {
     #[cfg(feature = "e2e-encryption")]
     pub async fn crypto_context_info(&self) -> CryptoContextInfo {
         let encryption = self.client.encryption();
+
+        let (device_creation_ts, this_device_is_verified) = match encryption.get_own_device().await
+        {
+            Ok(Some(device)) => {
+                (device.first_time_seen_ts(), device.is_verified_with_cross_signing())
+            }
+
+            // Should not happen, there will always be an own device
+            _ => (MilliSecondsSinceUnixEpoch::now(), true),
+        };
+
+        let is_backup_configured = encryption.backups().state() == BackupState::Enabled;
+        let backup_exists_on_server =
+            encryption.backups().exists_on_server().await.unwrap_or(false);
```

Comment:
it feels a bit odd that we would check if there is a backup on the server, even if the user has explicitly disabled backups? Maybe it's fine though.